sound/soc/sof/pm.c

```diff
@@ -246,6 +246,7 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 
 	/* reset FW state */
 	sdev->fw_state = SOF_FW_BOOT_NOT_STARTED;
+	sdev->enabled_cores_mask = 0;
```

Comment:
@lyakh why not tie this to the hda_dsp_core_power_down() function, so that you update this field whenever a core is powered up or down?

Reply:
@ranj063 in practice all `.core_power_up` operations are implemented by `hda_dsp_enable_core()` and `.core_power_down` by `hda_dsp_core_reset_power_down()` so yes, that would work. But in theory that doesn't have to be the case. `.enabled_cores_mask` is handles "above" DSP operations. You could being it down to operations, but then you'd also have to add `.core_power_state`. Do we want to do that?

Reply:
@lyakh Yeah, I tend to agree. enabled_cores_mask is managed in SOF core, so it would be weird to have state actions in Intel specific code. Simiarly the state transitions for sdev->fw_state are handled in SOF core, not in hda_dsp code.